Tests/Packet++Test/main.cpp

```diff
@@ -7399,6 +7400,35 @@ PTF_TEST_CASE(GtpLayerEditTest)
 	delete [] buffer2;
 }
 
+static void readFileAndValidateMessageType(int& ptfResult, const char *filename, BgpMessageType messageType)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	int buffer1Length = 0;
+	uint8_t* buffer1 = readFileIntoBuffer(filename, buffer1Length);
+	PTF_ASSERT_NOT_NULL(buffer1);
+
+	RawPacket rawPacket1((const uint8_t*)buffer1, buffer1Length, time, true);
+	Packet bgpPacket1(&rawPacket1);
+
+	PTF_ASSERT_TRUE(bgpPacket1.isPacketOfType(BGP));
+	BgpLayer* bgpLayer = bgpPacket1.getLayerOfType<BgpLayer>();
+	PTF_ASSERT_NOT_NULL(bgpLayer);
+
+	bgp_header* bgpHeader = bgpLayer->getBgpHeader();
+	PTF_ASSERT_NOT_NULL(bgpHeader);
+
+	PTF_ASSERT_TRUE(bgpHeader->messageType == messageType);

```

Comment:
Missing tests for:

- Header length
- Data that comes after the basic BGP header